dataset/test/bin_test.cpp

```diff
@@ -89,6 +90,16 @@ TEST_F(DataArrayBinTest, 2d) {
   EXPECT_EQ(bin(bin(table, {edges_x}), {edges_y}), bucketed);
 }
 
+TEST_F(DataArrayBinTest, operations_on_empty) {
+  const Variable indices = makeVariable<std::pair<scipp::index, scipp::index>>(
+      Dimensions{{Dim::Event, 0}, {Dim::Y, 0}}, Values{});
```

Comment:
It is confusing to have `Dim::Event` as both an external dimension as well as the internal bin dimension, use something else here?